scripts/http-jsonp-detection.nse

```diff
@@ -0,0 +1,188 @@
+          report = "Partially controllable from URL"
+        end
+      end
+    end
+  end
+  return report
+end
+
+--The function tries to bruteforce through the most common callback variable
+local callback_bruteforce = function(host, port, target)
+  local response, path, report
+  for _,p in ipairs(callbacks) do 
+    path = target
+    path = path .. "?" .. p .. "=test"
+    response = http.get(host, port, path)
+    if response and response.body and response.status and response.status==200 then 
+
+      local status, func
+      status, func = checkjson(response.body)
+
+      if status == true then  
+        report = callback_url(host, port, target, p)
+        if report ~= nil then 
+          report = string.format("%s\t%s", target, report)
+        else
+          report = target
+        end
+        break
+      end
+    end
+  end 
+  return report
+end 
+
+action = function(host, port)
+  local path = stdnse.get_script_args(SCRIPT_NAME .. ".path") or "/"
+  local output_xml = stdnse.output_table()
+  output_xml = {}
+  output_xml['jsonp-endpoints'] = {}
+  local output_str = "\nThe following JSONP endpoints were detected: " 
+
+  -- crawl to find jsonp endpoints urls
+  local crawler = httpspider.Crawler:new(host, port, path, {scriptname = SCRIPT_NAME})
+
+  if (not(crawler)) then
+    return
+  end
+
+  crawler:set_timeout(10000)
+
+  while(true) do
+    local status, r = crawler:crawl()
+    if (not(status)) then
+      if (r.err) then
+        return stdnse.format_output(false, r.reason)
+      else 
+        break
+      end
+    end
+
+    local target = tostring(r.url)
+    target = url.parse(target)
+    target = target.path
+
+    -- First we try to get the response and look for jsonp endpoint there 
+    if r.response and r.response.body and r.response.status and r.response.status==200 then
+
+      local status, func, report
+      status, func = checkjson(r.response.body)
+
+      if status == true then
+        --We have found JSONP endpoint
+        --Put it inside a returnable table.
+        output_str = string.format("%s\n%s", output_str, target) 
+        table.insert(output_xml['jsonp-endpoints'], target)
+
+        --Try if the callback function is controllable from URL.     
+        report = callback_url(host, port, target)        
+        if report ~= nil then
+          output_str = string.format("%s\t%s", output_str, report)
+        end
+
+      else 
+
+        --Try to bruteforce through most comman callback URLs
+        report = callback_bruteforce(host, port, target)
+        if report ~= nil then
+          table.insert(output_xml['jsonp-endpoints'], target)
+          output_str = string.format("%s\n%s", output_str, report)
+        end
+      end 
+
+    end 
+
+  end
+
+  --A way to print returnable 
+  if next(output_xml['jsonp-endpoints']) then 
+    return output_xml, output_str
+  else
+    output_str = "\nCouldnt find any JSONP endpoints"
+    stdnse.verbose(1, output_str)
```

Comment:
Use a nmap.verbosity() to determine if verbose is enabled, then return the negative response. This will help users who wonder if the script failed. By default we don't return anything but if verbosity is set we let the users know we couldn't find any endpoints.